core/arch/riscv64/emit_utils.c

```diff
@@ -340,8 +340,35 @@ stub_is_patched(dcontext_t *dcontext, fragment_t *f, cache_pc stub_pc)
 void
 unpatch_stub(dcontext_t *dcontext, fragment_t *f, cache_pc stub_pc, bool hot_patch)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    /* If stub_pc is not aligned to 4 bytes, the first instruction will be c.nop, see
+     * insert_exit_stub_other_flags(). */
+    stub_pc = ALIGNED(stub_pc, 4) ? stub_pc : stub_pc + 2;
+    /* At any time, at most one patching strategy will be in effect: the one for
+     * intermediate fragments or the one for far fragments.
+     */
+    if (stub_is_patched_for_intermediate_fragment_link(dcontext, stub_pc)) {
+        /* Restore the sd a0, offs(reg_stolen), see insert_exit_stub_other_flags().
+         * Format of the sd instruction:
+         *  | imm[11:5] |  rs2  |  rs1  |011| imm[4:0] |0100011|
+         *  ^   31-25   ^ 24-20 ^ 19-15 ^   ^   11-7   ^
+         */
+        ASSERT(TLS_REG0_SLOT <= (2 << 11) - 1);
+        *(uint *)vmcode_get_writable_addr(stub_pc) =
+            (0x3023 | TLS_REG0_SLOT >> 5 << 25 | (DR_REG_A0 - DR_REG_ZERO) << 20 |
```

Comment:
please add a #define for 0x3023, and add comment to describe what it is.

Reply:
The comment above should explain what 0x3023 is. If I were to add a #define, what should I call it?

Reply:
I'd call it something like SD_INSTR_MASK. I'll leave it to you. It's fine to leave it as is.